source/cl/test/UnitCL/source/cl_intel_unified_shared_memory/usm_event_info.cpp

```diff
@@ -99,155 +77,128 @@ TEST_F(USMEventInfoTest, clEnqueueMemFillINTEL_EventInfo) {
-                              sizeof(pattern) * 2, 0, nullptr, &wait_events[0]);
-  EXPECT_SUCCESS(err);
-
-  ASSERT_SUCCESS(GetEventInfoHelper(
-      wait_events[0], CL_EVENT_COMMAND_EXECUTION_STATUS, CL_QUEUED));
-
-  ASSERT_SUCCESS(clWaitForEvents(1, &wait_events[0]));
-
-  ASSERT_SUCCESS(GetEventInfoHelper(
-      wait_events[0], CL_EVENT_COMMAND_EXECUTION_STATUS, CL_COMPLETE));
-
-  ASSERT_SUCCESS(GetEventInfoHelper(wait_events[0], CL_EVENT_COMMAND_TYPE,
-                                    CL_COMMAND_MEMFILL_INTEL));
-
-  if (host_ptr) {
-    err =
-        clEnqueueMemFillINTEL(queue, host_ptr, pattern, sizeof(pattern),
-                              sizeof(pattern) * 2, 0, nullptr, &wait_events[1]);
+  for (auto ptr : allPointers()) {
+    cl_event wait_event;
+    err = clEnqueueMemFillINTEL(queue, ptr, pattern, sizeof(pattern),
+                                sizeof(pattern) * 2, 0, nullptr, &wait_event);
     EXPECT_SUCCESS(err);
 
     ASSERT_SUCCESS(GetEventInfoHelper(
-        wait_events[1], CL_EVENT_COMMAND_EXECUTION_STATUS, CL_QUEUED));
+        wait_event, CL_EVENT_COMMAND_EXECUTION_STATUS, CL_QUEUED));
 
-    ASSERT_SUCCESS(clWaitForEvents(1, &wait_events[1]));
+    ASSERT_SUCCESS(clWaitForEvents(1, &wait_event));
 
     ASSERT_SUCCESS(GetEventInfoHelper(
-        wait_events[1], CL_EVENT_COMMAND_EXECUTION_STATUS, CL_COMPLETE));
+        wait_event, CL_EVENT_COMMAND_EXECUTION_STATUS, CL_COMPLETE));
 
-    ASSERT_SUCCESS(GetEventInfoHelper(wait_events[1], CL_EVENT_COMMAND_TYPE,
+    ASSERT_SUCCESS(GetEventInfoHelper(wait_event, CL_EVENT_COMMAND_TYPE,
                                       CL_COMMAND_MEMFILL_INTEL));
-  }
 
-  EXPECT_SUCCESS(clReleaseEvent(wait_events[0]));
-  if (host_ptr) {
-    EXPECT_SUCCESS(clReleaseEvent(wait_events[1]));
+    EXPECT_SUCCESS(clReleaseEvent(wait_event));
   }
 }
 
 TEST_F(USMEventInfoTest, clEnqueueMemcpyINTEL_EventInfo) {
-  std::array<cl_event, 2> events;
-  void *offset_device_ptr = getPointerOffset(device_ptr, sizeof(cl_int));
-  cl_int err =
-      clEnqueueMemcpyINTEL(queue, CL_TRUE, offset_device_ptr, device_ptr,
-                           sizeof(cl_int), 0, nullptr, &events[0]);
-  EXPECT_SUCCESS(err);
-
-  ASSERT_SUCCESS(GetEventInfoHelper(
-      events[0], CL_EVENT_COMMAND_EXECUTION_STATUS, CL_COMPLETE));
-
-  ASSERT_SUCCESS(GetEventInfoHelper(events[0], CL_EVENT_COMMAND_TYPE,
-                                    CL_COMMAND_MEMCPY_INTEL));
-
-  if (host_ptr) {
-    err = clEnqueueMemcpyINTEL(queue, CL_FALSE, device_ptr, host_ptr,
-                               sizeof(cl_int), 0, nullptr, &events[1]);
-    EXPECT_SUCCESS(err);
-
-    ASSERT_SUCCESS(GetEventInfoHelper(
-        events[1], CL_EVENT_COMMAND_EXECUTION_STATUS, CL_QUEUED));
+  std::array<std::pair<void *, void *>, 4> pairs = {{
+      {host_ptr, device_ptr},
+      {host_ptr, shared_ptr},
+      {device_ptr, host_ptr},
+      {shared_ptr, host_ptr},
```

Comment:
If it makes sense to test types in both orders (this has both `{host_ptr, device_ptr}` and `{device_ptr, host_ptr}`), does not make sense to test both orders for all types?